src/pages-and-resources/discussions/app-list/AppList.test.jsx

```diff
@@ -1,21 +1,124 @@
+    store.dispatch(updateStatus({ status: LOADED }));
+    // renders component
+    const component = createComponent();
+    const wrapper = render(component);
+    container = wrapper.container;
+    // assert test case results
+    expect(queryByText(container, 'There are no discussions providers available for your course.')).toBeInTheDocument();
+  });
 
+  test('displays loading state when there is no selected App', () => {
+    // render component without mocking data in store
+    const component = createComponent();
+    const wrapper = render(component);
+    container = wrapper.container;
+    expect(queryByText(container, 'Loading...')).toBeInTheDocument();
   });
 
-  test('hides the FeaturesTable at mobile sizes', () => {
+  test('display a card for each available app', async () => {
+    // mock data in redux store
+    axiosMock.onGet(getAppsUrl(courseId)).reply(200, piazzaApiResponse);
+    await executeThunk(fetchApps(courseId), store.dispatch);
+    // render componenet
+    const component = createComponent();
+    const wrapper = render(component);
+    container = wrapper.container;
+    // assert test results
+    const appCount = store.getState().discussions.appIds.length;
+    const nodeCount = queryAllByRole(container, 'radio').length;
+    expect(nodeCount === appCount);
+  });
 
+  test('displays the FeaturesTable at desktop sizes', async () => {
+    axiosMock.onGet(getAppsUrl(courseId)).reply(200, piazzaApiResponse);
+    await executeThunk(fetchApps(courseId), store.dispatch);
+    const component = createComponent();
+    const wrapper = render(component);
+    container = wrapper.container;
+    expect(queryByRole(container, 'table')).toBeInTheDocument();
   });
 
-  test('onSelectApp is called when an app is clicked', () => {
+  /** Implementation needed to be done in App list component for hiding table on mobile screens
+   test('hides the FeaturesTable at mobile sizes', () => {
+    axiosMock.onGet(getAppsUrl(courseId)).reply(200, piazzaApiResponse);
+    await executeThunk(fetchApps(courseId), store.dispatch);
+    const component = createComponent(500);
+    const wrapper = render(component);
+    container = wrapper.container;
+    expect(screen.queryByRole('table')).not.toBeInTheDocument();
+  }); */
 
+  test('onSelectApp is called when an app is clicked', async () => {
+    // default  mocked data selected legacy app
+    axiosMock.onGet(getAppsUrl(courseId)).reply(200, legacyApiResponse);
+    await executeThunk(fetchApps(courseId), store.dispatch);
+    const component = createComponent();
+    const wrapper = render(component);
+    container = wrapper.container;
+    const uncheckedCard = getByRole(container, 'radio', { checked: false });
+    expect(queryByLabelText(uncheckedCard, 'Select Pizza'));
+    userEvent.click(uncheckedCard);
+    const checkedCard = getByRole(container, 'radio', { checked: true });
+    expect(queryByLabelText(checkedCard, 'Select Pizza'));
```

Comment:
how can check and unchecked input radio has sample label `Select Pizza`?

Reply:
have you missed calling `toBeInTheDocument()` on the expect? 

=> `expect(queryByLabelText(checkedCard, 'Select Pizza')).toBeInTheDocument();`

Reply:
this implementation has some faults I am working on fixing it as we discussed.